### PR TITLE
feature/1776 - Update requirements.txt and requirements-test.txt to reflect currently used dependencies

### PR DESCRIPTION
**Pull request by sasha-dresden**
Issue [FECFILE-1776](https://fecgov.atlassian.net/browse/FECFILE-1776)

Also after the dev meetup it's believed django storages is no longer required, so I removed it from the INSTALLED_APPS list. This didn't cause any breaking changes locally, and shouldn't once it goes to dev, but just something to keep an eye out for.

### CODE REVIEW

**Review by dheitzer — CHANGES_REQUESTED** [review]
Briefly checking it seems the following are not used:
```
retry
rich
django-otp
```
Seems that deptry is not picking up these (and maybe other) unused dependencies - I might suggest checking for them manually.

Also, would we want to take this opportunity to upgrade some of the requirements-test.txt at least? For example, flake8 is quite out of date.

**Review by lbeaufort — CHANGES_REQUESTED** [review]
I'd like to keep our builds more deterministic, and keep our pinned patch versions. ie
`==` instead of `~=`

File: requirements.txt [hunk 1]
```diff
@@ -1,35 +1,22 @@
...
-openapi-codec==1.3.2
-psycopg2-binary==2.9.9
-retry==0.9.2
-django-otp==1.5.4
-zeep==4.2.1
-django-structlog[celery]==8.1.0
-rich==13.8.1
-jwcrypto==1.5.6
-
-# Pinning sqlparse to 0.5.0 as it is a sub-dependency that needs to be upgraded for security
```
[the remark this answers is not in the document]
Comment (sasha-dresden) [reply requirements.txt:29]: Updated so everything is == instead of ~=

**Comment by sasha-dresden** [comment]
Closing. This will be handled as part of [PR1244](https://github.com/fecgov/fecfile-web-api/pull/1244) which does this and sets up the deptry circle task.

**Pull request closed without merging**